docs/Reference/bsp-v25.md

```diff
@@ -0,0 +1,284 @@
+| `MAX_MAP_FACES` |					1048576 |
+| `MAX_MAP_LEAFFACES` |				1048576 |
+| `MAX_MAP_LEAFBRUSHES` | 			1048576 |
+| `MAX_MAP_PORTALS` |					1048576 |
+| `MAX_MAP_CLUSTERS` |				1048576 |
+| `MAX_MAP_LEAFWATERDATA` |			32768 |
+| `MAX_MAP_PORTALVERTS` |				128000 |
+| `MAX_MAP_EDGES` |					4096000 |
+| `MAX_MAP_SURFEDGES` |				8192000 |
+| `MAX_MAP_LIGHTING` |				`0x1000000` |
+| `MAX_MAP_VISIBILITY` |				`0x1000000` |
+| `MAX_MAP_TEXTURES` |				16384 |
+| `MAX_MAP_CUBEMAPSAMPLES` |			16384 |
+| `MAX_MAP_OVERLAYS` |				16384 |
+| `MAX_MAP_WATEROVERLAYS` |			16384 |
+| `MAX_MAP_TEXDATA_STRING_DATA` |		256000 |
+| `MAX_MAP_TEXDATA_STRING_TABLE` |	1048576 |
+| `MAX_MAP_PRIMITIVES` |				1048576 |
+| `MAX_MAP_PRIMVERTS` |				1048576 |
+| `MAX_MAP_PRIMINDICES` |				1048576 |
+
+## Lumps
+
+In BSPv25, the lump header structure was modified to incorporate a version field. 
+Lump version starts at 0 and is defaulted to 0 for lumps that aren't versioned.
+
+```c
+struct lumpfileheader_t
+{
+	int				lumpOffset;
+	int				lumpID;
+	int				lumpVersion;	// Added by BSP v25
+	int				lumpLength;
+	int				mapRevision;
+};
+```
+
+### Lump Versions
+
+The following table lists current lump versions for all lumps that are versioned in Chaos:
+
+| Lump Name | Version |
+|---|---|
+| `LUMP_LIGHTING_VERSION` | 1 |
+| `LUMP_OCCLUSION_VERSION` | 2 |
+| `LUMP_LEAF_AMBIENT_LIGHTING_VERSION`  1 |
+| `LUMP_WORLDLIGHTS_VERSION` | 1 |
+| `LUMP_PHYSDISP_VERSION` | 2 |
+| `LUMP_NODES_VERSION` | 1 |
+| `LUMP_DISPINFO_VERSION` | 1 |
+| `LUMP_EDGES_VERSION` | 1 |
+| `LUMP_PRIMITIVES_VERSION` | 1 |
+| `LUMP_FACES_VERSION` | 2 |
+| `LUMP_FACEBRUSHES_VERSION` | 1 |
+| `LUMP_FACEBRUSHLIST_VERSION` | 1 |
+| `LUMP_LEAF_AMBIENT_INDEX_VERSION` | 1 |
+| `LUMP_LEAFS_VERSION` | 2 |
+| `LUMP_BRUSHSIDES_VERSION` | 1 |
+| `LUMP_AREAPORTALS_VERSION` | 1 |
+| `LUMP_LEAFFACES_VERSION` | 1 |
+| `LUMP_LEAFBRUSHES_VERSION` | 1 |
+| `LUMP_PRIMINDICES_VERSION` | 1 |
+| `LUMP_VERTNORMALINDICES_VERSION` | 1 |
+| `LUMP_OVERLAYS_VERSION` | 1 |
+| `LUMP_WATEROVERLAYS_VERSION` | 1 |
+| `LUMP_LEAFWATERDATA_VERSION` | 1  |
+
+## Structures
+
+In the BSP v25 update, all lumps listed in the lump version table have been modified. 
+This section lists the new structure format for each of those lumps.
+
+```cpp
+struct dphysdisp_t
+{
+	unsigned int numDisplacements;
+};
+
+struct dnode_t
+{
+	DECLARE_BYTESWAP_DATADESC();
+	int				planenum;
+	int				children[2];
+	float			mins[3];
+	float			maxs[3];
+	unsigned int	firstface;
+	unsigned int	numfaces;
+	short			area;
+};
+
+class ddispinfo_t
+{
+	Vector		startPosition;
+	int			m_iDispVertStart;
+	int			m_iDispTriStart;
+    int         power;
+    int         minTess;
+    float       smoothingAngle;
+    int         contents;
+	unsigned int	m_iMapFace;
+	int			m_iLightmapAlphaStart;
+	int			m_iLightmapSamplePositionStart;
+	CDispNeighbor<unsigned int>			m_EdgeNeighbors[4];
+	CDispCornerNeighbors<unsigned int>	m_CornerNeighbors[4];
```

Comment:
CDispNeighbor and CDispCornerNeighbors are not originally template types, do we want to have here an equivalent non-template type here?

Reply:
Changed these to generic struct types now